pallets/allocations/src/benchmarking.rs

```diff
@@ -26,37 +26,51 @@ use super::*;
 use crate::Pallet as Allocations;
 use frame_benchmarking::impl_benchmark_test_suite;
 use frame_benchmarking::{account, benchmarks};
+use frame_support::BoundedVec;
 use frame_system::RawOrigin;
 use sp_std::prelude::*;
 
-const MAX_BYTES: u32 = 1_024;
 const SEED: u32 = 0;
 
 pub struct BenchmarkConfig<T: Config> {
 	grantee: T::AccountId,
 	oracle: T::AccountId,
 }
 
-fn make_benchmark_config<T: Config>(u: u32) -> BenchmarkConfig<T> {
-	let grantee = account("grantee", u, SEED);
-	let oracle = account("oracle", u, SEED);
+fn make_benchmark_config<T: Config>() -> BenchmarkConfig<T> {
+	let grantee = account("grantee", 0, SEED);
+	let oracle = account("oracle", 0, SEED);
 
-	let deposit_applying = T::ExistentialDeposit::get();
+	BenchmarkConfig { grantee, oracle }
+}
 
-	T::Currency::make_free_balance_be(&grantee, deposit_applying);
-	T::Currency::make_free_balance_be(&oracle, deposit_applying);
+type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+fn make_batch<T: Config>(b: u32) -> BoundedVec<(T::AccountId, BalanceOf<T>), T::MaxAllocs> {
+	let mut ret = Vec::new();
 
-	BenchmarkConfig { grantee, oracle }
+	for i in 0..b {
+		let account = account("grantee", i, SEED);
+		ret.push((account, T::ExistentialDeposit::get() * 10u32.into()));
+	}
+
+	ret.try_into().unwrap()
 }
 
 benchmarks! {
 	allocate {
-		let b in 1 .. MAX_BYTES;
+		let config = make_benchmark_config::<T>();
+
+		Pallet::<T>::initialize_members(&[config.oracle.clone()]);
+	}: _(RawOrigin::Signed(config.oracle.clone()), config.grantee.clone(), T::ExistentialDeposit::get() * 10u32.into(), vec![])
+
+	batch {
+		let b in 1..T::MaxAllocs::get();
```

Comment:
When a range starts at 1 and does not include the last element it looks suspicious, is this intended here?

Reply:
actually a batch len of 0, we'd throw an error since there are no allocations to perform

Reply:
And the last to call should be `T::MaxAllocs::get()` or `T::MaxAllocs::get()-1`